lib/mpl_toolkits/tests/test_mplot3d.py

```diff
@@ -1713,13 +1713,33 @@ def test_margins():
     assert ax.margins() == (0, 0.1, 0)
 
 
-def test_margins_errors():
-    fig = plt.figure()
-    ax = fig.add_subplot(projection='3d')
-    with pytest.raises(TypeError, match="Cannot pass"):
-        ax.margins(0.2, x=0.4, y=0.4, z=0.4)
-    with pytest.raises(TypeError, match="Must pass"):
-        ax.margins(0.2, 0.4)
+@pytest.mark.parametrize('err, args, kwargs, match',
+                         ((ValueError, (-1,), {},
+                           'margin must be greater than -0.5'),
+                          (ValueError, (1, -1, 1), {},
+                           'margin must be greater than -0.5'),
+                          (ValueError, (1, 1, -1), {},
+                           'margin must be greater than -0.5'),
+                          (ValueError, tuple(), {'x': -1},
+                           'margin must be greater than -0.5'),
+                          (ValueError, tuple(), {'y': -1},
+                           'margin must be greater than -0.5'),
+                          (ValueError, tuple(), {'z': -1},
+                           'margin must be greater than -0.5'),
+                          (TypeError, (1, ), {'x': 1},
+                           'Cannot pass both positional and keyword'),
+                          (TypeError, (1, ), {'x': 1, 'y': 1, 'z': 1},
+                           'Cannot pass both positional and keyword'),
+                          (TypeError, (1, ), {'x': 1, 'y': 1},
+                           'Cannot pass both positional and keyword'),
+                          (TypeError, (1, 1), {},
+                           'Must pass a single positional argument for all*'),
+                          ))
```

Comment:
Can make this a little shorter:
```suggestion
@pytest.mark.parametrize('err, args, kwargs, match', (
    (ValueError, (-1,), {}, r'margin must be greater than -0\.5'),
    (ValueError, (1, -1, 1), {}, r'margin must be greater than -0\.5'),
    (ValueError, (1, 1, -1), {}, r'margin must be greater than -0\.5'),
    (ValueError, tuple(), {'x': -1}, r'margin must be greater than -0\.5'),
    (ValueError, tuple(), {'y': -1}, r'margin must be greater than -0\.5'),
    (ValueError, tuple(), {'z': -1}, r'margin must be greater than -0\.5'),
    (TypeError, (1, ), {'x': 1}, 'Cannot pass both positional and keyword'),
    (TypeError, (1, ), {'x': 1, 'y': 1, 'z': 1},
     'Cannot pass both positional and keyword'),
    (TypeError, (1, ), {'x': 1, 'y': 1},
     'Cannot pass both positional and keyword'),
    (TypeError, (1, 1), {}, 'Must pass a single positional argument for all*'),
))
```
Also, might want to escape the matches.